Submission receipt@0.1.0

edsu7 opened the pull request:
To address:
https://github.com/icgc-argo/argo-data-submission/issues/52

The following tests and their scenarios:
|Test|Scenario|
|--|--|
|local-test-badAnalysis.json| Analysis does not exist in the supplied database|
|local-test-badConnection.json| API request failed or URL is not correct|
|local-test-badFile.json| A file supplied was not found to exist within the specified analysis|
|local-test-successfulQuery.json| Successful query should return a `${analysis_id}_submission_receipt.tsv` populated with submitterIds, argoIds, fileName, analysisId, publishing time|

edsu7 committed:
[wfpm v0.8.0] added starting template for submission-receipt@0.1.0

edsu7 committed:
initial commit

edsu7 committed:
rename test file

- moved `test-successfulQuery.json` to `local-test-successfulQuery.json`. Github doesn't have access to QA

lindaxiang commented on submission-receipt/main.py:
```diff
@@ -0,0 +1,103 @@
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-o', '--output_file', dest='output_file',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    args = parser.parse_args()
```
`help` information are all the same? :-)

edsu7 replied:
Fixed help descriptions.

lindaxiang commented on submission-receipt/main.py:
```diff
@@ -0,0 +1,103 @@
+            writer.writeheader()
+            for file in response.json()['files']:
+                if file['fileName'] not in args.files:
+                    sys.exit("Specified file %s was not found in analysis %s" % (file['fileName'],args.analysis_id))
```
This will report the files in the `analysis` which are not in the the supplied  `args.files`? So it can't report 
`Specified file was not found in analysis`

edsu7 replied:
Fixed this line. Added checks for analysis files in supplied and vice versa.

https://github.com/icgc-argo/argo-data-submission/blob/5232a5bc7df766cc56cfa011b138e9c2a3b0b509/submission-receipt/main.py#L80-L85

lindaxiang replied:
These cross checks may not be very necessary. Since the submission has completed, the purpose of this receipt is to let the user know what we have received. The idea of cross checks is good, however it will introduce some complexity in the overall workflow in providing the `List of files to verify against analysis ID` especially when the submitted files are CRAM.
How do you think? @edsu7

edsu7 replied:
Added a `skip-check` as a compromise. If unnecessary check can be skipped.

https://github.com/icgc-argo/argo-data-submission/blob/e0737f8113ccdeee015179b73af6cb4fe02ddb5e/submission-receipt/main.py#L89-L95

lindaxiang commented on submission-receipt/main.py:
```diff
@@ -0,0 +1,103 @@
+                "objectId":None,
+                "fileName":None,
+                "fileMd5sum":None
+                }
```
Do we consider the scenario when the submitted files are original CRAMs?

edsu7 replied:
For `CRAM`, the receipt will display the values for converted `BAM`. Will that be enough? Technically the `CRAM` format was not submitted.

lindaxiang replied:
When the users submitted `CRAM`, we can add additional fields `original_fileName`, `original_fileSize` and `original_fileMd5sum` since the user only have info for their submitted `CRAM` for cross check.

edsu7 replied:
added:
https://github.com/icgc-argo/argo-data-submission/blob/e0737f8113ccdeee015179b73af6cb4fe02ddb5e/submission-receipt/main.py#L115-L118

edsu7 committed:
update `main.py`

- fixed file descriptions
- clarified loop when looking for supplied argument files in analysis
- added loop for when looking for analysis files in supplied argument

lindaxiang commented on submission-receipt/main.py:
```diff
@@ -0,0 +1,107 @@
+                "studyId":args.study_id,
+                "objectId":file['objectId'],
+                "fileName":file['fileName'],
+                "fileMd5sum":file['fileMd5sum']
```
Please add `fileSize` as well.

edsu7 replied:
Added:
https://github.com/icgc-argo/argo-data-submission/blob/e0737f8113ccdeee015179b73af6cb4fe02ddb5e/submission-receipt/main.py#L112

edsu7 committed:
updates to main.nf

- add flag to skip checks
- added cram file related fields

edsu7 committed:
update build-test

edsu7 committed:
update typo

lindaxiang reviewed (approved):
LGTM
